File: clients/php/tests/pop.php
<?php
use RedisPriorityQueue as RQP;

// Requires
require_once(__DIR__.'/../utils/redis_instance.php');
require_once(__DIR__.'/../utils/const.php');
require_once(__DIR__.'/../lib/Lua.php');
require_once(__DIR__.'/../lib/Queue.php');

// Create Queue instance with Redis connection
$q = new RQP\Queue($r->getConnection());

// Set queue name
$q->setQueueName(QUEUE_NAME);

// Run
echo "* pop (one):\n";
$res = $q->popOne();
echo $res."\n";

// Run
echo "* pop (one, descending):\n";
$res = $q->popOne('desc');
echo $res."\n";

// Run
echo "* pop (many):\n";
$res = $q->popMany('asc', MULTIPLE_ITEMS_COUNT);
echo json_encode($res)."\n";

File: clients/python/pop.py
#!/usr/bin/env python3

# Packages
import redis
from lib.RpqQueue import RpqQueue
from lib.RpqLua import RpqLua
from utils.redis_vars import *

def listItems(items):
    '''Helper to print items'''
    for item in items:
        print (item);

# Redis instance
# !!! set Redis credentials into utils/redis_vars.py
r = redis.StrictRedis(host=rHost, port=rPort, db=rDbnum, password=rAuth)

# Load LUA Script
RpqLua = RpqLua()
RpqLua.setRedisConnection(r);
RpqLua.loadSource('../../src/redis-priority-queue.lua');
queue = RpqLua.register();

# RpqQueue instance
RpqQueue = RpqQueue()

# Set LUA Script
RpqQueue.setLuaScript(queue)

# Set queue name
RpqQueue.setqueueName('simple_queue')

# Pop an item / some items

print ('* pop (one):')
item = RpqQueue.popOne()
if item:
    print (item);
else:
    print ('Queue is empty');

print ('* pop (one, ascending):')
item = RpqQueue.popOne('asc')
if item:
    print (item);
else:
    print ('Queue is empty');

print ('* pop (many):')
items = RpqQueue.popMany('desc', 5)
if items:
    listItems(items)
else:
    print ('Queue is empty');
